### PR TITLE
Added info about Open forms from other bindings, and updated bindings requirements. (Part 2)

### PR DESCRIPTION
**Pull request by larkox**
#### Summary
As in https://github.com/mattermost/mattermost-developer-documentation/pull/932, I piggybacked two different changes on this PR:
- Open forms from any location. Now, if a binding for embedded bindings has a form, it will open directly a modal form.
- Binding filtering. We added several filters to forms and bindings, to make sure they are well formed. Updated the documentation to take those in consideration.

#### Related PRs
Open form in any location
https://github.com/mattermost/mattermost-mobile/pull/5665
https://github.com/mattermost/mattermost-webapp/pull/8784
Filtering:
https://github.com/mattermost/mattermost-webapp/pull/8782
https://github.com/mattermost/mattermost-mobile/pull/5655

### CODE REVIEW

File: site/content/integrate/apps/interactivity/_index.md [hunk 1]
```diff
@@ -37,15 +37,15 @@ Markdown fields are a special field that allows you to better format your form.
...
-| `name`        | string    | Key to use in the values field of the call.                              |
-| `type`        | FieldType | The type of the field.                                                   |
-| `is_required` | bool      | (Optional) Whether the field needs to be filled.                         |
-| `multiselect` | bool      | (Optional) Whether a select field allows multiple values to be selected. |
-| `value`       | value     | (Optional) Default value.                                                |
-| `description` | string    | (Optional) Text to show below the field describing it.                   |
-| `modal_label` | string    | (Optional) Label to name the field in the modal.                         |
+| Name          | Type      | Description                                                                                                 |
+| :------------ | :-------- | :---------------------------------------------------------------------------------------------------------- |
+| `name`        | string    | Key to use in the values field of the call. Cannot include spaces nor tabs. Cannot include spaces nor tabs. |
```
Comment (justinegeffen) [inline site/content/integrate/apps/interactivity/_index.md:42]:
```suggestion
| `name`        | string    | Key to use in the values field of the call. Cannot include spaces nor tabs. |
```

File: site/content/integrate/apps/interactivity/_index.md [hunk 2]
```diff
@@ -137,28 +137,31 @@ Bindings are of two types, buttons or selects.
...
-| `location` | string | Location name. The whole location path will be provided in the context. |
-| `label`    | string | Label that will show in the button.                                     |
-| `call`     | Call   | Call to be made when the button is clicked.                             |
+| Name       | Type   | Description                                                                                                                                          |
+| :--------- | :----- | :---------------------------------------------------------------------------------------------------------------------------------------- |
+| `location` | string | Location name. The whole location path will be provided in the context.                                                                              |
+| `label`    | string | Label that will show in the button. Defaults to location. Must be unique in its level.                                                                |
+| `call`     | Call   | (Optional) Call to be made when the button is clicked. You must provide a Call if there is no Form, or the Form itself does not have a Call.          |
```
Comment (justinegeffen) [inline site/content/integrate/apps/interactivity/_index.md:144]:
```suggestion
| `call`     | Call   | (Optional) Call to be made when the button is selected. You must provide a call if there is no form, or the form itself does not have a call.          |
```

File: site/content/integrate/apps/interactivity/_index.md [hunk 3]
```diff
@@ -137,28 +137,31 @@ Bindings are of two types, buttons or selects.
...
+| `form`     | Form   | (Optional) Form to open in a modal form when the button is clicked. You must provide a Form with a Call if there is no Call defined in the Binding. |
```
Comment (justinegeffen) [inline site/content/integrate/apps/interactivity/_index.md:145]:
```suggestion
| `form`     | Form   | (Optional) Form to open in a modal form when the button is clicked. You must provide a form with a call if there is no call defined in the binding. |
```

File: site/content/integrate/apps/interactivity/_index.md [hunk 4]
```diff
@@ -137,28 +137,31 @@ Bindings are of two types, buttons or selects.
...
-| `label`    | string | User-facing string.                                                                                                |
-| `call`     | Call   | (Optional) Call to perform when the option is selected. If none is defined, it will take the call from the select. |
+| Name       | Type   | Description                                                                                                                                          |
+| :--------- | :----- | :----------------------------------------------------------------------------------------------------------------------------------------- |
+| `location` | string | Option name. The whole location path will be provided in the context.                                                                                |
+| `label`    | string | User-facing string. Defaults to location. Must be unique in its level.                                                                                |
+| `call`     | Call   | (Optional) Call to perform when the option is selected. You must provide a Call if there is no Form, or the Form itself does not have a Call.    |
```
Comment (justinegeffen) [inline site/content/integrate/apps/interactivity/_index.md:163]:
```suggestion
| `call`     | Call   | (Optional) Call to perform when the option is selected. You must provide a call if there is no form, or the form itself does not have a call.    |
```

**Review by justinegeffen — COMMENTED** [review]
Thanks, @larkox! I've made some suggestions. The reason I suggested we change "call", "form", and "binding" to lowercase is because that's how we write it in the documentation (https://developers.mattermost.com/integrate/apps/call/ and https://developers.mattermost.com/integrate/apps/bindings/) so it's just for consistency.

LMK if you have any questions!

File: site/content/integrate/apps/interactivity/_index.md [hunk 5]
```diff
@@ -37,15 +37,15 @@ Markdown fields are a special field that allows you to better format your form.
...
+| :------------ | :-------- | :---------------------------------------------------------------------------------------------------------- |
+| `name`        | string    | Key to use in the values field of the call. Cannot include spaces nor tabs. Cannot include spaces nor tabs. |
+| `type`        | FieldType | The type of the field.                                                                                      |
+| `is_required` | bool      | (Optional) Whether the field needs to be filled.                                                            |
+| `multiselect` | bool      | (Optional) Whether a select field allows multiple values to be selected.                                    |
+| `value`       | value     | (Optional) Default value.                                                                                   |
+| `description` | string    | (Optional) Text to show below the field describing it.                                                      |
+| `modal_label` | string    | (Optional) Label to name the field in the modal. Defaults to `label` (if defined) or `name`.                |
```
Comment (hanzei) [inline site/content/integrate/apps/interactivity/_index.md:48]: What takes priority?

**Review by hanzei — APPROVED** [review]

File: site/content/integrate/apps/interactivity/_index.md (hunk 5 above)
Comment (larkox) [reply site/content/integrate/apps/interactivity/_index.md:48]: Label if defined, if not, name.

**Review by justinegeffen — APPROVED** [review]

**Pull request merged**